### PR TITLE
stabilize liveliness API

### PR DESCRIPTION
**Pull request by milyin**
Stablilization of liveliness API

### CODE REVIEW

**Review by DenisBiryukov91 — APPROVED** [review]

File: zenoh/src/api/session.rs [hunk 1]
```diff
@@ -2352,6 +2340,7 @@ impl SessionInner {
         self.task_controller
             .spawn_with_rt(zenoh_runtime::ZRuntime::Net, {
                 let session = WeakSession::new(self);
+                #[cfg(feature = "unstable")]
```
[the remark this answers is not in the document]
Comment (milyin) [reply zenoh/src/api/session.rs]: done

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -143,15 +143,15 @@ jobs:
...
 
       - name: Run tests with SHM
         if: ${{ matrix.os == 'macOS-latest' || matrix.os == 'windows-latest' }}
-        run: cargo nextest run -F test -F shared-memory -F unstable -E 'not (test(test_default_features))' --exclude zenoh-examples --exclude zenoh-plugin-example --workspace
+        run: cargo nextest run -F test -F shared-memory -F unstable -F internal_config -E 'not (test(test_default_features))' --exclude zenoh-examples --exclude zenoh-plugin-example --workspace
 
       - name: Run tests with SHM + unixpipe
         if: ${{ matrix.os == 'ubuntu-latest' }}
-        run: cargo nextest run -F test -F shared-memory -F unstable -F transport_unixpipe -E 'not (test(test_default_features))' --exclude zenoh-examples --exclude zenoh-plugin-example --workspace
+        run: cargo nextest run -F test -F shared-memory -F unstable -F internal_config -F transport_unixpipe -E 'not (test(test_default_features))' --exclude zenoh-examples --exclude zenoh-plugin-example --workspace
```
Comment (Mallets) [inline .github/workflows/ci.yml:154]: Why there is the need of enabling `internal_config`?

File: zenoh-ext/examples/Cargo.toml [hunk 3]
```diff
@@ -34,7 +34,7 @@ default = []
 [dependencies]
 tokio = { workspace = true, features = ["rt", "sync", "time", "macros", "io-std"] }
 futures = { workspace = true }
-zenoh = { workspace = true, features = ["unstable"], default-features = false }
+zenoh = { workspace = true, features = ["unstable", "internal_config"], default-features = false }
```
Comment (Mallets) [inline zenoh-ext/examples/Cargo.toml:37]: Why there is the need of enabling `internal_config`?

Comment (milyin) [reply zenoh-ext/examples/Cargo.toml:37]: zenoh-ext uses direct access to config fields. Before it was hidden as the "unstable" included "internal_config"
Later I think we need to remove this feature https://github.com/eclipse-zenoh/zenoh/issues/1655

File: .github/workflows/ci.yml (hunk 2 above)
Comment (milyin) [reply .github/workflows/ci.yml:154]: The liveliness test (and maybe others) uses direct access to config which is under this feature. I wanted to enable zenoh tests in stable mode so I had to detach `internal_config` from `unstable`
https://github.com/eclipse-zenoh/zenoh/issues/1654

**Comment by Mallets** [comment]
@milyin is there an umbrella issue keeping track of API stabilization all repositories?

**Comment by milyin** [comment]
> @milyin is there an umbrella issue keeping track of API stabilization all repositories?

https://github.com/eclipse-zenoh/zenoh/issues/1645

**Review by Mallets — APPROVED** [review]

**Pull request merged**